cli/azd/cmd/provision.go

```diff
@@ -125,6 +129,32 @@ func (p *provisionAction) Run(ctx context.Context) (*actions.ActionResult, error
 		defaultTitleNote = "This is a preview. No changes will be applied to your Azure resources."
 	}
 
+	// Get Subscription to Display in Command Title Note
+	subscriptions, subErr := p.subManager.GetSubscriptions(ctx)
+	if subErr == nil {
+		messageFormat := "Provisioning Azure resources in subscription (%s) %s and location (%s) can take some time"
+		if previewMode {
+			messageFormat = "This is a preview. No changes will be applied to your Azure resources in subscription (%s) %s " +
+				"and location (%s)."
+		}
```

Comment:
nit. Move this inside the `if sub.Id == p.env.GetSubscriptionId() {` as that's the only case when this is used.